compiler/rustc_target/src/callconv/s390x.rs

```diff
@@ -32,19 +36,25 @@ where
         }
         return;
     }
-    if !arg.layout.is_aggregate() && arg.layout.size.bits() <= 64 {
+
+    let size = arg.layout.size;
+    if size.bits() <= 128 && arg.layout.is_single_vector_element(cx) {
```

Comment:
There's one more quirk I think we need to handle here: for single-element vector aggregates, the aggregate must have no padding, i.e. the aggregate size must match the vector size.   (This is different for FP aggregates, where e.g. an 8-byte aligned struct with a single float element is passed like a double.)

Reply:
Ah, good catch. Older versions of this PR had [a check for that](
https://github.com/rust-lang/rust/commit/730ecfe194e6046bd03b94ba2f73937d6a9929d2#diff-3292be2ad8ef8734ebff20f7b377fa7691af9607e9ba55b4725733b5cdded003R12), but I accidentally removed it when I rewrote it based on is_single_fp_element.

Re-added size check and added a test for this case:

https://github.com/rust-lang/rust/blob/1bd141663b9b4f488f45b2d9b26e18423a722e68/tests/assembly/s390x-vector-abi.rs#L265-L271